### PR TITLE
GUACAMOLE-176: Support layer location and opacity for guac_common_surface

### PR DESCRIPTION
**Pull request by mike-jumper**

### CODE REVIEW

File: src/common/common/surface.h [hunk 1]
```diff
@@ -317,10 +363,77 @@ void guac_common_surface_clip(guac_common_surface* surface, int x, int y, int w,
...
+ */
+void guac_common_surface_stack(guac_common_surface* surface, int z);
+
+/**
+ * Changes the parent layer of ths given surface. By default, layers will be
+ * children of the default layer. If the surface does not represent a
+ * non-default visible layer, then this function has no effect.
+ *
+ * @param surface
+ *     The surface to move relative to its parent layer.
```
Comment (jmuehlner) [inline src/common/common/surface.h]: Looks like maybe some bad copypasta here? I don't think this parameter description works here, or in guac_common_surface_set_opacity.

Comment (mike-jumper) [inline src/common/common/surface.h]: Oops.

File: src/common/common/surface.h [hunk 2]
```diff
@@ -317,10 +363,77 @@ void guac_common_surface_clip(guac_common_surface* surface, int x, int y, int w,
...
+ */
+void guac_common_surface_set_parent(guac_common_surface* surface,
+        const guac_layer* parent);
+
+/**
+ * Sets the opacity of the surface. If the surface does not represent a
+ * non-default visible layer, then this function has no effect.
+ *
+ * @param surface
+ *     The surface to move relative to its parent layer.
```
Comment (jmuehlner) [inline src/common/common/surface.h]: To move? Doesn't this function change opacity?

Comment (mike-jumper) [inline src/common/common/surface.h]: Sigh ... yet more copypasta ... 🍝

**Pull request merged**